debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresDefaultValueConverter.java

```diff
@@ -0,0 +1,177 @@
+
+        result.put("bpchar", v -> extractDefault(v, "")); // Sample value: 'abcd'::bpchar
+        result.put("varchar", v -> extractDefault(v, "")); // Sample value: `abcde'::character varying
+        result.put("text", v -> extractDefault(v, "")); // Sample value: 'asdf'::text
+
+        result.put("numeric", v -> new BigDecimal(extractDefault(v, "0.0"))); // Sample value: 12345.67891
+        result.put("float4", v -> Float.parseFloat(extractDefault(v, "0.0"))); // Sample value: 1.234
+        result.put("float8", v -> Double.parseDouble(extractDefault(v, "0.0"))); // Sample values: `1.234`, `'12345678901234567890'::numeric`
+        result.put("int2", v -> Short.parseShort(extractDefault(v, "0"))); // Sample value: 32767
+        result.put("int4", v -> Integer.parseInt(extractDefault(v, "0"))); // Sample value: 123
+        result.put("serial", v -> Integer.parseInt(extractDefault(v, "0")));
+        result.put("int8", v -> Long.parseLong(extractDefault(v, "0"))); // Sample values: `123`, `'9223372036854775807'::bigint`
+        result.put("bigserial", v -> Long.parseLong(extractDefault(v, "0")));
+
+        result.put("json", v -> extractDefault(v, "{}")); // Sample value: '{}'::json
+        result.put("jsonb", v -> extractDefault(v, "{}")); // Sample value: '{}'::jsonb
+        result.put("xml", v -> extractDefault(v, "")); // Sample value: '<foo>bar</foo>'::xml
+
+        result.put("uuid", v -> UUID.fromString(extractDefault(v, "00000000-0000-0000-0000-000000000000"))); // Sample value: '76019d1a-ad2e-4b22-96e9-1a6d6543c818'::uuid
+
+        result.put("date", v -> timestampUtils.toLocalDateTime(extractDefault(v, "1970-01-01")));
+        result.put("time", v -> timestampUtils.toLocalTime(extractDefault(v, "00:00")));
+        result.put("timestamp", v -> timestampUtils.toOffsetDateTime(extractDefault(v, "1970-01-01")));
+        result.put("timestamptz", v -> timestampUtils.toOffsetDateTime(extractDefault(v, "1970-01-01")));
+        result.put("interval", v -> new PGInterval(extractDefault(v, "epoch")));
+
+        // Other data types, such as box, bytea, and more are not handled.
+        return result;
+    }
+
+    private static String extractDefault(String defaultValue) {
+        // Values are either "raw", such as `1234`, or "type casted", such as `'9223372036854775807'::bigint`.
+        // If the value does NOT contain a single quote it is assumed to be a raw value. Otherwise the value is
+        // extracted from inside the single quotes.
+        if (!defaultValue.contains("'")) {
+            return defaultValue;
+        }
+
+        final Matcher matcher = LITERAL_DEFAULT_PATTERN.matcher(defaultValue);
+        matcher.find();
+        return matcher.group(1);
+    }
+
+    // If the default value is generated by a function, map a placeholder value for the schema
+    private static String extractDefault(String defaultValue, String generatedValuePlaceholder) {
```

Comment:
This extends the original implementation to exit early with a 'placeholder' value if the default value is supplied by a function. It's a simplistic approach, so let me know if there are any further cases which should be handled specially